src/main/java/org/reactome/release/common/dataretrieval/FileRetriever.java

```diff
@@ -0,0 +1,364 @@
+	 */
+	public FileRetriever(String retrieverName)
+	{
+		//logger.trace("Setting retrieverName to {}", retrieverName);
+		this.setRetrieverName(retrieverName);
+		this.logger = this.createLogger(this.retrieverName, "RollingRandomAccessFile", this.getClass().getName(), true, Level.DEBUG, this.logger, "Data Retriever");
+	}
+	
+	@Override
+	public void fetchData() throws Exception 
+	{
+		if (this.uri == null)
+		{
+			throw new RuntimeException("You must provide a URI from which the file will be downloaded!");
+		}
+		else if (this.destination.trim().length() == 0)
+		{
+			throw new RuntimeException("You must provide a destination to which the file will be downloaded!");
+		}
+		//Before fetching anything, we need to check to see if the file already exists.
+		Path pathToFile = Paths.get(this.destination);
+		if (Files.exists(pathToFile))
+		{
+			BasicFileAttributes attributes = Files.readAttributes(pathToFile, BasicFileAttributes.class);
+			
+			Instant fileCreateTime = attributes.creationTime().toInstant();
+			Instant now = Instant.now();
+			//If the file is older than the maxAge...
+			if (fileCreateTime.isBefore( now.minus(this.maxAge) ))
+			{
+				//TODO: Option to save back-ups of old files.
+				logger.debug("File {} is older than allowed amount ({}) so it will be downloaded again.",this.destination,this.maxAge);
+				downloadData();
+				logger.debug("Download is complete.");
+			}
+			else
+			{
+				logger.debug("File {} is not older than allowed amount ({}) so it will not be downloaded.",this.destination,this.maxAge);
+			}
+		}
+		else
+		{
+			logger.debug("File {} does not exist and must be downloaded.", pathToFile);
+			//if file does not exist, get it from the URL.
+			downloadData();
+		}
+		
+		// Print some basic file stats.
+		if (Files.exists(pathToFile))
+		{
+			if (Files.isReadable(pathToFile))
+			{
+				BasicFileAttributes attribs = Files.readAttributes(pathToFile, BasicFileAttributes.class);
+				logger.info("File Info: Name: {}, Size: {}, Created: {}, Modified: {}",this.destination,attribs.size(), attribs.creationTime(), attribs.lastModifiedTime());
+			}
+			else
+			{
+				logger.error("File {} is not readable!", pathToFile);
+			}
+		}
+		else
+		{
+			// If something failed during the data retrieval, the file might not exist. If that happens, let the user know.
+			logger.error("File \"{}\" still does not exist after executing the file retriever!", pathToFile);
+		}
+	}
+
+	protected void downloadData() throws Exception
+	{
+		logger.trace("Scheme is: {}", this.uri.getScheme());
+		Path path = null;
+		try
+		{
+			path = Paths.get(this.destination);
+			Files.createDirectories(path.getParent());
+			if (this.uri.getScheme().equals("http") || this.uri.getScheme().equals("https"))
+			{
+				
+				doHttpDownload(path);
+			}
+			else if (this.uri.getScheme().equals("ftp") || this.uri.getScheme().equals("sftp"))
```

Comment:
Similar here: `this.uri.getScheme().endsWith("ftp")` (though I'm okay leaving it as is in this case since I always forget that the "s" comes at the start for secure ftp!

Reply:
Any thoughts on this?

Reply:
It works as it is. May change later but not interested in re-testing right now.